### PR TITLE
Build both minified and unminified assets; utilize SCRIPT_DEBUG constant

### PR DESCRIPTION
**Pull request by philipnewcomer**
This PR includes the following, in regards to #19
* Update task runners to build both minified and unminified assets
* Utilize the SCRIPT_DEBUG constant to reference unminified assets if SCRIPT_DEBUG is enabled
* Move login CSS to a separate task so we're not building it on every compile

@jacobarriola since this touches the frontend workflow I'd like to get your feedback on this before I merge it.

cc @AaronHolbrook

### CODE REVIEW

**Comment by jacobarriola** [comment]
Thanks for this @philipnewcomer ! 🌮 

What's the value or purpose in producing an unminified version of an asset?

**Comment by philipnewcomer** [comment]
@jacobarriola unminified assets are important for debugging. It's difficult to debug a minified Javascript file, so using the SCRIPT_DEBUG constant provides an easy way to use unminified assets in our local and staging environments where most in-browser debuging happens.

I think I just included unminified CSS out of habit, as unminified CSS was important before sourcemaps became widely used and I got used to always having both minified and unminifed versions of all assets. A valid argument could be made that with sourcemaps, there is no need for unminified CSS.

It also looks like sourcemaps are a thing for Javascript now too. I will investigate whether Javascript sourcemaps would eliminate the need for unminified Javascript files.

**Comment by philipnewcomer** [comment]
@jacobarriola From my investigations, I have come to the conclusion that CSS sourcemaps remove the need for an unminified version of the CSS for use during development. Thus, I have removed the generation of unminified CSS from this PR.

Javascript sourcemaps are also a thing, and they more or less work, but in my testing they did not provide enough resolution to enable the setting of breakpoints in all the places that an unminified Javascript file would. I also ran into some bugs with the sourcemaps for minified Javascript files becoming mangled and inaccurate, which seems to be an issue that the frontend community is still working through. Thus, I have concluded that having an unminified version of the theme's scripts **are** useful for development, until the state of Javascript sourcemaps improves.

To summarize, this PR now does the following:
- Provide both a minified and unminified version of the `main` and `foundation` Javascript files
- Utilizes the `SCRIPT_DEBUG` constant to determine which to load
- Renames the compiled CSS file from `app.css` to `app.min.css` to be consistent with the Javascript file naming
- Minifies the compiled login CSS file and creates a sourcemap for it

Let me know your thoughts on this updated PR.
